### PR TITLE
37 jshint fixes

### PR DESCRIPTION
**Pull request by lucyfox4131**
#### What's this PR do?
- Fix JSHint complaint regarding both missing and unnecessary semicolons at the end of lines or functions
- Fix additional JSHint complaint regarding assignment statement within edit
#### Where should the reviewer start?
- All errors were fixed within JS files exclusively. Application/cable JS were not touched, start with Delete.js 
#### How should this be manually tested?
- Tests can be run to ensure all are passing. 
#### Any background context you want to provide?
- Only fix was JS hint errors 
#### What gif best describes this PR or how it makes you feel?

giphy

@rrgayhart, @ckaminer

### CODE REVIEW

File: spec/model/idea_spec.rb [hunk 1]
```diff
@@ -1,15 +0,0 @@
-require 'rails_helper'
-
-RSpec.describe Idea, type: :model do
```
Comment (rrgayhart) [inline spec/model/idea_spec.rb:3]: Any reason this test is getting deleted? Was it failing?

Comment (rrgayhart) [inline spec/model/idea_spec.rb:3]: OH JUST KIDDING I GET IT

**Comment by rrgayhart** [comment]
Nice work!

File: spec/model/idea_spec.rb (hunk 1 above)
Comment (lucyfox4131) [reply spec/model/idea_spec.rb:3]: Those generators get me some times. I think what happened here was I had a duplicate spec and one had info the other didn't so I just put it all together

File: app/assets/javascripts/delete.js [hunk 2]
```diff
@@ -16,4 +14,9 @@ function deleteItem (id){
       console.log("Success");
     }
   });
-};
+  removeRow(id);
+}
+
```
Comment (ckaminer) [inline app/assets/javascripts/delete.js:19]: I like this decision to make removeRow a reusable function.

File: app/assets/javascripts/filter.js [hunk 3]
```diff
@@ -6,14 +6,14 @@ function searchFilter(){
     $(".idea-row").each(function(index, idea){
       var title = $(this).find("td:first").text().toLowerCase();
       var body = $(this).find("td:nth-child(2)").text().toLowerCase();
-      var ideaOrBodyContains = title.indexOf(currentSearch) !== -1 || body.indexOf(currentSearch) !== -1
+      var ideaOrBodyContains = title.indexOf(currentSearch) !== -1 || body.indexOf(currentSearch) !== -1;
```
Comment (ckaminer) [inline app/assets/javascripts/filter.js:9]: team semi-colon!!!

**Pull request merged**